### PR TITLE
Dependency updates + Detekt

### PR DESCRIPTION
**Pull request by Maxr1998**

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
     kotlin("android")
     kotlin("kapt")
     id("kotlin-parcelize")
+    id("io.gitlab.arturbosch.detekt") version Dependencies.Versions.detekt
     id("de.mannodermaus.android-junit5")
     id("com.github.ben-manes.versions") version Dependencies.Versions.dependencyUpdates
 }
 
+detekt {
+    buildUponDefaultConfig = true
```
Comment (nielsvanvelzen) [inline app/build.gradle.kts:16]: Adding `ignoreFailures = true` will make sure Detekt reports all issues - handy for when we add linting to CI.

File: app/build.gradle.kts [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+    allRules = false
+    config = files("${rootProject.projectDir}/detekt.yml")
+
+    reports {
+        html.enabled = true
+        xml.enabled = false
+        txt.enabled = true
```
Comment (nielsvanvelzen) [inline app/build.gradle.kts:23]: Do we need txt reports? :p

File: app/build.gradle.kts [hunk 3]
```diff
@@ -145,31 +157,45 @@ dependencies {
...
 }
 
-tasks.withType<Test> {
-    useJUnitPlatform()
-    testLogging {
-        outputs.upToDateWhen { false }
-        showStandardStreams = true
+tasks {
+    withType<Detekt> {
+        jvmTarget = JavaVersion.VERSION_1_8.toString()
```
Comment (nielsvanvelzen) [inline app/build.gradle.kts:162]: I don't think this is needed?

File: app/build.gradle.kts (hunk 2 above)
Comment (Maxr1998) [reply app/build.gradle.kts:23]: They're nice to read and the overhead is zero, so sure.

File: app/build.gradle.kts (hunk 1 above)
Comment (Maxr1998) [reply app/build.gradle.kts:16]: Theoretically, I want to have all failure fixed. I'll start working on it once ExoPlayer is merged.

File: app/build.gradle.kts (hunk 2 above)
Comment (nielsvanvelzen) [reply app/build.gradle.kts:23]: Fair enough

File: app/build.gradle.kts (hunk 3 above)
Comment (Maxr1998) [reply app/build.gradle.kts:162]: Detekt recommends it in the setup. Idk.

File: app/build.gradle.kts (hunk 1 above)
Comment (nielsvanvelzen) [reply app/build.gradle.kts:16]: Yes but we want all the issues reported to GitHub so therefor it should be ignored.

File: app/build.gradle.kts (hunk 3 above)
Comment (nielsvanvelzen) [reply app/build.gradle.kts:162]: I didn't need it for other projects I've used Detekt and it works fine for those

File: app/build.gradle.kts (hunk 1 above)
Comment (Maxr1998) [reply app/build.gradle.kts:16]: Ok :+1:

File: app/build.gradle.kts (hunk 3 above)
Comment (Maxr1998) [reply app/build.gradle.kts:162]: Afaik it serves as a hint to detekt. So not 'needed', but might be helpful?

Comment (nielsvanvelzen) [reply app/build.gradle.kts:163]: https://detekt.github.io/detekt/gradle.html#using-type-resolution

The default `detekt` task already uses Java 1.8

Comment (Maxr1998) [reply app/build.gradle.kts:163]: I can't read that from there, but I'll remove it ig.

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**